### PR TITLE
Adds option to reset curl cainfo

### PR DESCRIPTION
**Pull request by JohanMabille**
This is required by micromamba

cc @wolfv @Klaim

### CODE REVIEW

File: src/curl.cpp [hunk 1]
```diff
@@ -78,7 +78,11 @@ namespace powerloader
             if (verifystatus != CURLE_OK && verifystatus != CURLE_NOT_BUILT_IN)
                 throw curl_error("Could not initialize CURL handle");
 
-            if (!ctx.ssl_ca_info.empty())
+            if (ctx.ssl_erase_ca_info)
+            {
+                setopt<char*>(CURLOPT_CAINFO, nullptr);
+            }
+            else if (!ctx.ssl_ca_info.empty())
```
Comment (wolfv) [inline src/curl.cpp:85]: should they be connected with `else if`? Maybe there is a use case where you never want to use default, but still want to use a set ca_info?

**Review by wolfv — CHANGES_REQUESTED** [review]

File: src/curl.cpp (hunk 1 above)
Comment (JohanMabille) [reply src/curl.cpp:85]: In that case the user does not change `ssl_erase_ca_info` (which is false by default) and sets the `ssl_ca_info` to the value he needs.

 The way I wrote the code ensures that `ssl_erase_ca_info` has the highest priority.

Comment (wolfv) [reply src/curl.cpp:85]: hmm, but in potential user code they would now have to adjust two values if they want to set the ssl_ca_info to some deterministic value. E.g.

```
if (NEVER_USE_CURL_DEFAULT) {
   plctx.ssl_erase_ca_info = true;
}

if (env::get("CA_LOCATION")) {
   plctx.ssl_ca_info = env::get("CA_LOCATION");
   // now that would be a source of bugs even those two things are completely independent
   pltctx.ssl_erase_ca_info = false;
}

Comment (JohanMabille) [reply src/curl.cpp]: Indeed, I've updated the PR.

**Pull request merged**